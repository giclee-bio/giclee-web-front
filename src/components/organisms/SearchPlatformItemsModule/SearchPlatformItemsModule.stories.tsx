import { Meta, StoryObj } from '@storybook/react'

import SearchPlatformItemsModule from './SearchPlatformItemsModule'

const meta: Meta<typeof SearchPlatformItemsModule> = {}

export default meta
type Story = StoryObj<typeof SearchPlatformItemsModule>

export const Index: Story = {
  args: {},
}
