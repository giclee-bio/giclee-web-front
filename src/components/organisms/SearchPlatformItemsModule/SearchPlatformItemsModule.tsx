import { CaretRightIcon } from '@radix-ui/react-icons'

import SearchBar from '@/components/atoms/SearchBar'
import PlatformItemsList from '@/components/molecules/PlatformItemsList'

import type { PlatformItemProps } from '@/components/atoms/PlatformItem'

const SearchPlatformItemsModule = () => {
  const item = {
    alt: 'nacal',
    caption: 'nacal',
    src: '/nacal.png',
  }
  const items: PlatformItemProps[] = new Array(8).fill(item)

  return (
    <section className='max-w-md rounded-lg bg-stone-800'>
      <h2 className='pl-8 pt-6 font-bold'>新しいIDを追加</h2>
      <section className='flex justify-between p-8 pt-4'>
        <SearchBar />
        <button className='ml-8 flex items-center'>
          View All <CaretRightIcon className='ml-1 inline' height='16' width='16' />
        </button>
      </section>
      <section className=''>
        <PlatformItemsList items={items} />
      </section>
    </section>
  )
}

export default SearchPlatformItemsModule
