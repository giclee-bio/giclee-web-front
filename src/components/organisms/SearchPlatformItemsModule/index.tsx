import SearchPlatformItemsModule from './SearchPlatformItemsModule'

export default SearchPlatformItemsModule
