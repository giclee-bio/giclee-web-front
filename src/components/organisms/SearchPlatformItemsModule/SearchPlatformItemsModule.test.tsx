import { render } from '@testing-library/react'

import SearchPlatformItemsModule from './SearchPlatformItemsModule'

const items = [
  {
    alt: 'nacal',
    caption: 'nacal',
    platformAccountName: 'aaaa',
    src: '/nacal.png',
  },
  {
    alt: 'nacal2',
    caption: 'nacal2',
    platformAccountName: 'bbbb',
    src: '/nacal.png',
  },
  {
    alt: 'nacal3',
    caption: 'nacal3',
    platformAccountName: 'cccc',
    src: '/nacal.png',
  },
]

describe('SearchPlatformItemsModule', () => {
  it('snapshot', async () => {
    const { asFragment } = render(
      <SearchPlatformItemsModule handleSelectItem={() => jest.mock} items={items} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
