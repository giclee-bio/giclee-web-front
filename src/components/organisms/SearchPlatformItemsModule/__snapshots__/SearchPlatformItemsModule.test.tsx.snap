// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPlatformItemsModule snapshot 1`] = `
<DocumentFragment>
  <section
    class="max-w-md rounded-lg bg-stone-800"
  >
    <h2
      class="pl-8 pt-6 font-bold"
    >
      新しいIDを追加
    </h2>
    <section
      class="flex justify-between p-8 pt-4"
    >
      <form>
        <div>
          <label
            class="sr-only"
            for="radix-:r0:"
          >
            Search
          </label>
          <div
            class="relative flex h-8"
          >
            <svg
              class="absolute left-1 h-8 text-stone-400"
              fill="none"
              height="24"
              viewBox="0 0 15 15"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 6.5C10 8.433 8.433 10 6.5 10C4.567 10 3 8.433 3 6.5C3 4.567 4.567 3 6.5 3C8.433 3 10 4.567 10 6.5ZM9.30884 10.0159C8.53901 10.6318 7.56251 11 6.5 11C4.01472 11 2 8.98528 2 6.5C2 4.01472 4.01472 2 6.5 2C8.98528 2 11 4.01472 11 6.5C11 7.56251 10.6318 8.53901 10.0159 9.30884L12.8536 12.1464C13.0488 12.3417 13.0488 12.6583 12.8536 12.8536C12.6583 13.0488 12.3417 13.0488 12.1464 12.8536L9.30884 10.0159Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
            <input
              class="rounded bg-stone-50 pl-8 pr-3 text-black placeholder:text-sm placeholder:text-stone-200"
              id="radix-:r0:"
              name="search"
              placeholder="さがす"
              required=""
              title=""
              type="search"
            />
          </div>
        </div>
      </form>
      <button
        class="ml-8 flex items-center"
      >
        View All 
        <svg
          class="ml-1 inline"
          fill="none"
          height="16"
          viewBox="0 0 15 15"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.18194 4.18185C6.35767 4.00611 6.6426 4.00611 6.81833 4.18185L9.81833 7.18185C9.90272 7.26624 9.95013 7.3807 9.95013 7.50005C9.95013 7.6194 9.90272 7.73386 9.81833 7.81825L6.81833 10.8182C6.6426 10.994 6.35767 10.994 6.18194 10.8182C6.0062 10.6425 6.0062 10.3576 6.18194 10.1819L8.86374 7.50005L6.18194 4.81825C6.0062 4.64251 6.0062 4.35759 6.18194 4.18185Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </section>
    <section
      class=""
    >
      <div
        class="h-32 max-w-md overflow-hidden"
        dir="ltr"
        style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
      >
        <style>
          [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
        </style>
        <div
          class="h-full w-full rounded"
          data-radix-scroll-area-viewport=""
          style="overflow-x: scroll; overflow-y: hidden;"
        >
          <div
            style="min-width: 100%; display: table;"
          >
            <div
              class="flex"
            >
              <li
                class="ml-6 list-none first:ml-8"
              >
                <button
                  class="w-20"
                >
                  <div
                    class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
                  >
                    <img
                      alt="nacal"
                      class="rounded"
                      data-nimg="1"
                      decoding="async"
                      height="48"
                      loading="lazy"
                      src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                      srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                      style="color: transparent;"
                      width="48"
                    />
                  </div>
                  <p
                    class="mt-1 text-center text-sm"
                  >
                    nacal
                  </p>
                </button>
              </li>
              <li
                class="ml-6 list-none first:ml-8"
              >
                <button
                  class="w-20"
                >
                  <div
                    class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
                  >
                    <img
                      alt="nacal2"
                      class="rounded"
                      data-nimg="1"
                      decoding="async"
                      height="48"
                      loading="lazy"
                      src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                      srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                      style="color: transparent;"
                      width="48"
                    />
                  </div>
                  <p
                    class="mt-1 text-center text-sm"
                  >
                    nacal2
                  </p>
                </button>
              </li>
              <li
                class="ml-6 list-none first:ml-8"
              >
                <button
                  class="w-20"
                >
                  <div
                    class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
                  >
                    <img
                      alt="nacal3"
                      class="rounded"
                      data-nimg="1"
                      decoding="async"
                      height="48"
                      loading="lazy"
                      src="/_next/image?url=%2Fnacal.png&w=96&q=75"
                      srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
                      style="color: transparent;"
                      width="48"
                    />
                  </div>
                  <p
                    class="mt-1 text-center text-sm"
                  >
                    nacal3
                  </p>
                </button>
              </li>
            </div>
          </div>
        </div>
      </div>
    </section>
  </section>
</DocumentFragment>
`;
